### ISSUE TITLE
uv self update error on Windows: The system cannot move the file to a different disk drive

### ISSUE THREAD

**Issue by ThatOtherAndrew** [astral-sh/uv#3845]
I am using a [Dev Drive](https://learn.microsoft.com/en-us/windows/dev-drive/) on Windows 11, mounted at `D:`. As recommended by Microsoft docs, I have mapped CARGO_HOME to this drive to improve cache speeds. This seems to cause issues with `uv self update` however, resulting in the following error:

```
uv self update
info: Checking for updates...
error: The system cannot move the file to a different disk drive. (os error 17)
[Exit code 2]
```

The issue was reproduced on `uv` version 0.2.3, one older than the current version. The issue cannot be reproduced on the latest v0.2.4 as `uv self update` exits early upon detecting being on the latest version, however I am making the assumption that the issue is still present.

**Comment by charliermarsh** [comment]
This is likely an issue in https://github.com/axodotdev/axoupdater. Do you mind filing there? Otherwise I can.

**Comment by ThatOtherAndrew** [comment]
Forwarded, thank you for the quick response!

**Comment by charliermarsh** [comment]
Thanks! We'll keep this open in the interim.

**Comment by mistydemeo** [comment]
I've opened a PR on Axoupdater that should fix this (https://github.com/axodotdev/axoupdater/pull/121). Due to the US holiday, it'll most likely take us a day or two to test and cut a new release.

**Comment by ThatOtherAndrew** [comment]
Don't mean to unnecessarily bump this issue, but just for the record, I believe the upstream issue was fixed with [axoupdater v0.6.5](https://github.com/axodotdev/axoupdater/releases/tag/v0.6.5).

**Comment by charliermarsh** [comment]
Thank you yes! And we updated.
